### ISSUE TITLE
gpt2 io binding inference error when 'past_sequence_length' is 0

### ISSUE THREAD

**Issue by carter54** [microsoft/onnxruntime#6245]
**Describe the bug**
I followed this notebook to test the gpt2 inference with onnxruntime
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/notebooks/Inference_GPT2_with_OnnxRuntime_on_CPU.ipynb
following error is given when prepare_io_binding with empty_past

```
Traceback (most recent call last):
  File "/home/hr/PycharmProjects/test/inference_torch_onnx.py", line 254, in <module>
    test_generation(tokenizer, EXAMPLE_Text, ort_session=session)
  File "/home/hr/PycharmProjects/test/inference_torch_onnx.py", line 219, in test_generation
    outputs = inference_with_io_binding(ort_session, config, input_ids, position_ids, attention_mask, past)
  File "/home/hr/PycharmProjects/test/inference_torch_onnx.py", line 191, in inference_with_io_binding
    io_binding = Gpt2Helper.prepare_io_binding(session, input_ids, position_ids, attention_mask, past,
  File "/home/hr/anaconda3/lib/python3.8/site-packages/onnxruntime/transformers/gpt2_helper.py", line 404, in prepare_io_binding
    io_binding.bind_input(f'past_{i}', past_i.device.type, 0, float_type, list(past_i.size()),
  File "/home/hr/anaconda3/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 291, in bind_input
    self._iobinding.bind_input(name,
RuntimeError: /onnxruntime_src/onnxruntime/python/onnxruntime_pybind_state.cc:1333 onnxruntime::python::addObjectMethods(pybind11::module&, onnxruntime::Environment&)::<lambda(onnxruntime::SessionIOBinding*, const string&, const OrtDevice&, pybind11::object&, std::vector<long int>&, int64_t)> data_ptr != 0 was false. Pointer to data memory is not valid
```
This error also happened when I set
past_sequence_lengths = 0 in benchmark_gpt2.py
https://github.com/microsoft/onnxruntime/blob/111ac299cc26d0b7dfca68e500855e927653c157/onnxruntime/python/tools/transformers/benchmark_gpt2.py#L87

When I generate a dummy input with past_sequence_lengths > 0, the test worked fine.

**System information**
- OS Platform and Distribution: Linux Ubuntu 16.04
- ONNX Runtime installed from: pip install onnxruntime-gpu
- ONNX Runtime version: 1.6.0
- Python version: 3.8.5
- CUDA/cuDNN version: CUDA10.2, cuDNN8.0.3
- GPU model and memory: V100 16G

**To Reproduce**
set past_sequence_lengths = 0 in https://github.com/microsoft/onnxruntime/blob/111ac299cc26d0b7dfca68e500855e927653c157/onnxruntime/python/tools/transformers/benchmark_gpt2.py

**Comment by carter54** [comment]
#6204 report the same issue

**Comment by tianleiwu** [comment]
I can reproduce the issue on onnxruntime 1.6.0.  Onnxruntime 1.5.2 does not have the problem, so I think the cause is some recent change of IO binding API.

Before it is resolved,  you can use Onnxruntime 1.5.2 to walk around the issue.
